### PR TITLE
Unify waitpid handling

### PR DESCRIPTION
**Pull request by jfhg**

### CODE REVIEW

File: lib/pure/osproc.nim [hunk 1]
```diff
@@ -1014,28 +1013,30 @@ elif not defined(useNimRtl):
...
 
-      result = int(p.exitCode) shr 8
+      result = int(p.exitStatus) shr 8
   else:
     import times
 
     const
       hasThreadSupport = compileOption("threads") and not defined(nimscript)
 
+    var status : cint = 1
```
Comment (Araq) [inline lib/pure/osproc.nim]: A global variable? That's not re-entrant nor threadsafe.

**Comment by Araq** [comment]
What problems does this solve?

File: lib/pure/osproc.nim (hunk 1 above)
Comment (jfhg) [inline lib/pure/osproc.nim]: Urgh, should be fixed now.

**Comment by jfhg** [comment]
This PR tries to generally update p.exitCode only upon successful completion of waitpid().
In the current code this field is sometimes modified and then restored. Also in one case, the restore is only available under a when(freebsd) where as in other cases this is done generally.
This PR does not fix any bug I can reproduce. I came across the situation while debugging  #4792.

**Comment by jfhg** [comment]
Sorry, it took me a while... I now added a test case that fails in current devel (tested on Linux and DragonFly BSD). To fix it I also added some missing exitStatus to exitCode conversions.

**Pull request merged**